### ISSUE TITLE
Mint fails to build SwiftGen with Xcode 9.4.1

### ISSUE THREAD

**Issue by DannySe** [SwiftGen/SwiftGen#550]
Hi guys, 

I have come across the same issue as described in #457, but with version 6.0 and the libxml2 is installed. 

```
$ mint install SwiftGen/SwiftGen@6.0.0 --verbose
🌱  Cloning https://github.com/SwiftGen/SwiftGen.git 6.0.0...
Cloning into 'github.com_SwiftGen_SwiftGen'...
remote: Enumerating objects: 809, done.
remote: Counting objects: 100% (809/809), done.
remote: Compressing objects: 100% (621/621), done.
remote: Total 809 (delta 244), reused 419 (delta 140), pack-reused 0
Receiving objects: 100% (809/809), 8.69 MiB | 4.05 MiB/s, done.
Resolving deltas: 100% (244/244), done.
Note: checking out 'fa34f944de47bf986c957fae21ee263961894658'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

🌱  Building SwiftGen with SPM...
Failed to build SwiftGen
```
Switching to Xcode 10 seemed to fix the issue, but not for 6.0.2

```
$ mint install SwiftGen/SwiftGen@6.0.2 --verbose
Finding latest version of swiftgen
🌱  Resolved latest version of swiftgen to 6.0.2
🌱  Cloning https://github.com/SwiftGen/SwiftGen.git 6.0.2...
🌱  Building swiftgen with SPM...
🌱  Installing...
🌱  Copying resources for swiftgen: templates, Resources/Stencil-Info.plist, Resources/StencilSwiftKit-Info.plist, Resources/SwiftGen-Info.plist, Resources/SwiftGenKit-Info.plist ...
An error occurred: Error Domain=NSCocoaErrorDomain Code=4 "The file “Stencil-Info.plist” doesn’t exist." UserInfo={NSSourceFilePathErrorKey=/var/folders/vw/2yb4pyqd58gc79nv09vjsc6sghbsck/T/mint_560950335.206562/github.com_SwiftGen_SwiftGen/Resources/Stencil-Info.plist, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/usr/local/lib/mint/packages/github.com_SwiftGen_SwiftGen/build/6.0.2/Resources/Stencil-Info.plist, NSFilePath=/var/folders/vw/2yb4pyqd58gc79nv09vjsc6sghbsck/T/mint_560950335.206562/github.com_SwiftGen_SwiftGen/Resources/Stencil-Info.plist, NSUnderlyingError=0x7fea7651fd10 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

**Comment by AliSoftware** [comment]
Hi @DannySe 

Thanks for taking the time to file the issue

I don't think that's exactly the same issue as #457 (I mean, it's still the issue of building with Mint not working, but the reason isn't the same compilation error), it's rather a resource that seems missing.

What I don't understand at first glance is that according to the `--verbose` log you posted, the error seems to happen because `mint` can't find a Resource file (`Stencil-Info.plist`) while trying to copy it… but that's a task executed by mint, not by Xcode, so even if that was because we forgot to include that resource somehow, I'm not sure why the Xcode version (9.4 vs 10) would affect the outcome…

Anyway, we'll take a closer look soon, thanks reporting it!

**Comment by AliSoftware** [comment]
Mmmh just tested on my Mac with Xcode 10 selected and it seems to work fine:

```sh
$ mint install SwiftGen/SwiftGen
🌱  Finding latest version of SwiftGen
🌱  Resolved latest version of SwiftGen to 6.0.2
🌱  Cloning https://github.com/SwiftGen/SwiftGen.git 6.0.2...
🌱  Building SwiftGen Package with SPM...
🌱  Installing SwiftGen...
🌱  Copying resources for SwiftGen: templates, Resources/Stencil-Info.plist, Resources/StencilSwiftKit-Info.plist, Resources/SwiftGen-Info.plist, Resources/SwiftGenKit-Info.plist ...
🌱  Installed SwiftGen 6.0.2
🌱  An executable that was not installed by mint already exists at /usr/local/bin/swiftgen that is symlinked to /usr/local/Cellar/swiftgen/6.0.2/bin/swiftgen.
Ovewrite it with Mint's symlink? (y/n)
n

$ xcode-select -p
/Applications/Xcode-10.app/Contents/Developer

$ xcodebuild -version 
Xcode 10.0
Build version 10A255
```

@DannySe Are you sure you had Xcode 10 selected when you tried that `mint install SwiftGen/SwiftGen@6.0.2 --verbose`? What's the output of `xcode-select -p` and `xcodebuild -version` in those cases?

**Comment by AliSoftware** [comment]
Also, which version of `mint` are you using? (Mine is 0.11.2)

I see they fixed some issues with the installation of resources in Mint 0.11.0, maybe that's related?

**Comment by djbe** [comment]
Yeah SwiftGen probably won't build with Xcode 9.4 as we use Swift 4.2. 

The resource issue is a bit weird though, we purposefully tested this before release on our devices and it worked then. Curious.

**Comment by AliSoftware** [comment]
Given what's described in the Mint PR 
https://github.com/yonaskolb/Mint/pull/93 I say it's more probable to be an issue with Mint (the issue which has been fixed in Mint 0.11.0 since) than with SwiftGen itself

**Comment by djbe** [comment]
Right, so the original issue (won't build using Xcode 9.4) is a "won't fix", we upgraded to Swift 4.2 in #498 (SwiftGen 6.0.0).

The resources issue seems to be a Mint issue (as @AliSoftware mentioned above), which was fixed with Mint 0.11.

**Comment by DannySe** [comment]
You were right, updating mint to 0.11.2 and using Xcode 10 fixed the issues. Thanks for the quick reply!
